### PR TITLE
Actually "flatten UnionRDDs.

### PR DESCRIPTION
**Pull request by harveyfeng**
The code was there, it just didn't do anything...

### CODE REVIEW

File: src/main/scala/shark/execution/MemoryStoreSinkOperator.scala [hunk 1]
```diff
@@ -175,7 +175,6 @@ class MemoryStoreSinkOperator extends TerminalOperator {
             // of the current data and the SELECT output.
             hasPreviousRDDForUnion = true
             RDDUtils.unionAndFlatten(queryOutputRDD, previousRDD)
-            queryOutputRDD.union(previousRDD.asInstanceOf[RDD[TablePartition]])
```
Comment (harveyfeng) [inline src/main/scala/shark/execution/MemoryStoreSinkOperator.scala:178]: This was the buggy line :(

File: src/test/scala/shark/SQLSuite.scala [hunk 2]
```diff
@@ -667,6 +668,29 @@ class SQLSuite extends FunSuite with BeforeAndAfterAll {
       "StorageLevel for partition(keypart=2) should be NONE, but got: " + keypart2StorageLevel)
   }
 
+  ///////////////////////////////////////////////////////////////////////////////////////
+  // Prevent nested UnionRDDs - those should be "flattened" in MemoryStoreSinkOperator.
+  ///////////////////////////////////////////////////////////////////////////////////////
+
+  test("flatten UnionRDDs") {
+    def isFlattenedUnionRDD(unionRDD: UnionRDD[_]) = {
+      unionRDD.rdds.filter(_.isInstanceOf[UnionRDD[_]]).isEmpty
```
Comment (rxin) [inline src/test/scala/shark/SQLSuite.scala]: Use find - slightly more clear than filter.isEmpty

File: src/test/scala/shark/SQLSuite.scala [hunk 3]
```diff
@@ -667,6 +668,29 @@ class SQLSuite extends FunSuite with BeforeAndAfterAll {
...
+    }
+
+    sc.sql("insert into table test_cached select * from test")
```
Comment (rxin) [inline src/test/scala/shark/SQLSuite.scala]: Can you also add a test case for partitioned tables?

**Comment by harveyfeng** [comment]
Thanks for the speedy review! The Shark tests should pass now that I'm not modifying 'test_cached' in SQLSuite...

**Comment by rxin** [comment]
Thanks!

**Pull request merged**